src/frontend/vite.config.ts

```diff
@@ -51,8 +52,36 @@ export default defineConfig({
   base: '',
   build: {
     manifest: true,
-    outDir: '../../src/backend/InvenTree/web/static/web',
-    sourcemap: is_coverage
+    // outDir: '../../src/backend/InvenTree/web/static/web',
+    // TODO: outDir should be set to the correct path,
+    // TODO: depending on whether we are building library or UI?
+    sourcemap: true, // is_coverage,
+    lib: {
+      entry: {
+        components: resolve(__dirname, 'lib/components.ts'),
+        core: resolve(__dirname, 'lib/core.ts'),
+        functions: resolve(__dirname, 'lib/functions.ts'),
+        tables: resolve(__dirname, 'lib/tables.ts')
+      },
+      name: 'InvenTree',
+      formats: ['es']
+    },
+    rollupOptions: {
+      // TODO: This should be updated for all extrenal deps
+      external: ['React'],
```

Comment:
This is a tricky question. I would vote for trying to minimize the external dependencies for the exported components as much as possible and maybe only mark thinks like `react`, `@mantine/*`, `tanstack-query` as external.